htmlbeautifier: init at 1.3.1

mveytsman opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://hydra.nixos.org/job/nixpkgs/trunk/manual/latest/download/1/nixpkgs/manual.html#chap-reviewing-contributions
-->

###### Motivation for this change

Adds [htmlbeautifier](https://github.com/threedaymonk/htmlbeautifier) an HTML formatter with Ruby/ERB support.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

doronbehar commented:
Don't know why, but please note that there's a CI evaluation error.

shazow commented on maintainers/maintainer-list.nix:
```diff
@@ -5412,6 +5412,12 @@
     name = "Victor Collod";
   };
+  mveytsman = {
+    email = "maxim@ontoillogicmal.com";
+    github = "mvetsman";
+    githubId = 34720;
+    name = "Max Vetsman";
```
```suggestion
    email = "maxim@ontoillogical.com";
    github = "mveytsman";
    githubId = 34720;
    name = "Max Veytsman";
```

mveytsman replied:
<3 thanks @shazow

doronbehar commented on maintainers/maintainer-list.nix:
```diff
@@ -5412,6 +5412,12 @@
     githubId = 5047140;
     name = "Victor Collod";
   };
+  mveytsman = {
```
You also need to make sure your PR eventually has 2 commits:

```
maintainers: add mveytsman
htmlbeautifier: init at 1.3.1
```

mveytsman replied:
Thanks, that wasn't clear to me, I amended the history accordingly

doronbehar commented on pkgs/development/web/htmlbeautifier/default.nix:
```diff
@@ -0,0 +1,21 @@
+  exes = [ "htmlbeautifier" ];
+
+  # toString prevents the update script from being copied into the nix store
+  passthru.updateScript = toString ./update;
```
Question: Do all bundlerApp packages use such an update script? The script it self seems trivial that I doubt there is no generic way of doing so. It seems there is _something_ related to that according to https://nixos.org/nixpkgs/manual/#sec-language-ruby

mveytsman replied:
I found similar ones in [cocoapods](https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/mobile/cocoapods/update) but I see some that don't have it, e.g [ruby-zoom](https://github.com/NixOS/nixpkgs/tree/master/pkgs/tools/text/ruby-zoom) and [solargraph](https://github.com/NixOS/nixpkgs/tree/master/pkgs/development/ruby-modules/solargraph).

Do you think I should remove it?

doronbehar replied:
I think cocoapods' update script is a just a bit "non-trivial" because it also handles a Gemfile-beta file for as far as I can see.

Can you try:

```nix
  passthru.updateScript = bundlerUpdateScript "htmlbeautifier";
```

You should be able to check if this works with:

```sh
nix-shell maintainers/scripts/update.nix --argstr package htmlbeautifier
```

See https://nixos.org/nixpkgs/manual/#var-passthru-updateScript

mveytsman replied:
@doronbehar ah amazing! thanks for clarifying. I changed it as you suggested and it seems to work.

kimat replied:
> Do all bundlerApp packages use such an update script?

Most of "simple ruby packages" do, since: https://github.com/NixOS/nixpkgs/pull/64822.

mveytsman committed:
htmlbeautifier: init at 1.3.1

doronbehar reviewed (approved):
LGTM but I'm no experienced in reviewing bundler PRs. Maybe @aanderse will have something more to add.

aanderse commented:
Sorry, I can't be of help... I'm not a `ruby` guy :disappointed:

mveytsman commented:
No worries!

Do you know if there's anyone else we should ask for a :eyes: on this PR?

kimat commented:
I verified that:

- `nixpkgs-review pr 87928` works
- `[nix-shell:~/.cache/nixpkgs-review/pr-87928]$ htmlbeautifier --help` works
- the application works using : `[nix-shell:~/.cache/nixpkgs-review/pr-87928]$ htmlbeautifier < a.html > b.html` with sample files from https://github.com/threedaymonk/htmlbeautifier/blob/master/spec/documents_spec.rb

aanderse commented:
Well... I hate to do it, but I'll bug @manveru for a quick review here.

manveru commented on pkgs/development/web/htmlbeautifier/default.nix:
```diff
@@ -0,0 +1,20 @@
+    description = "A normaliser/beautifier for HTML that also understands embedded Ruby, ideal for tidying up Rails templates";
+    homepage = "https://github.com/threedaymonk/htmlbeautifier";
+    license = licenses.mit;
+    platforms = platforms.linux;
```
This should probably be `ruby.meta.platforms`, since it's pure Ruby code and should run everywhere Ruby does?

manveru commented on pkgs/development/web/htmlbeautifier/default.nix:
```diff
@@ -0,0 +1,20 @@
+bundlerApp {
+  inherit ruby;
+  pname = "htmlbeautifier";
+  gemfile = ./Gemfile;
+  lockfile = ./Gemfile.lock;
+  gemset = ./gemset.nix;
```
This can be expressed as `gemdir = ./.;`

manveru reviewed (changes requested):
Two small issues, looks good otherwise.

mveytsman committed:
maintainers: add mveytsman

mveytsman commented:
@manveru I'm really sorry I somehow missed this PR was still open with feedback from you until now :frowning_face: 

I went back and addressed your feedback. Is this good to be accepted now?

Sorry again for the hiatus!

SuperSandro2000 commented on pkgs/development/web/htmlbeautifier/gemset.nix:
```diff
@@ -0,0 +1,12 @@
+    };
+    version = "1.3.1";
+  };
+}
```
Missing final new line.

SuperSandro2000 commented on pkgs/development/web/htmlbeautifier/default.nix:
```diff
@@ -0,0 +1,18 @@
+    homepage = "https://github.com/threedaymonk/htmlbeautifier";
+    license = licenses.mit;
+    platforms = ruby.meta.platforms;
+    maintainers =  with maintainers; [mveytsman];
```
```suggestion
    maintainers =  with maintainers; [ mveytsman ];
```

SuperSandro2000 commented:
@ofborg eval

doronbehar reviewed (changes requested):
your PR should have eventually two commits:

```
maintainers: add <me>
<package-name>: init at <version>
```

doronbehar commented:
@mveytsman the PR looks good, only the commit log is not as it should be. There should be only two commits:

```
maintainers: add <me>
<package-name>: init at <version>
```

mveytsman commented:
Fixed!

doronbehar commented:
Error when testing the binary:

```
(ins)[nix-shell:~/.cache/nixpkgs-review/pr-87928]$ ./results/htmlbeautifier/bin/htmlbeautifier --help
Traceback (most recent call last):
        3: from ./results/htmlbeautifier/bin/htmlbeautifier:19:in `<main>'
        2: from /nix/store/w25m3pyvypghdi0shx1lg6z1d4g5lhz0-bundler-2.2.24/lib/ruby/gems/2.7.0/gems/bundler-2.2.24/lib/bundler.rb:149:in `setup'
        1: from /nix/store/w25m3pyvypghdi0shx1lg6z1d4g5lhz0-bundler-2.2.24/lib/ruby/gems/2.7.0/gems/bundler-2.2.24/lib/bundler/runtime.rb:13:in `setup'
/nix/store/w25m3pyvypghdi0shx1lg6z1d4g5lhz0-bundler-2.2.24/lib/ruby/gems/2.7.0/gems/bundler-2.2.24/lib/bundler/definition.rb:426:in `ensure_equivalent_gemfile_and_lockfile': You are trying to install in deployment mode after changing (Bundler::ProductionError)
your Gemfile. Run `bundle install` elsewhere and add the
updated ../../../../../nix/store/csw60zvq0rlf0494k86a598hyrh1jjl8-gemfile-and-lockfile/Gemfile.lock to version control.

If this is a development machine, remove the /nix/store/csw60zvq0rlf0494k86a598hyrh1jjl8-gemfile-and-lockfile/Gemfile freeze
by running ``.

The list of sources changed

You have added to the Gemfile:
* source: locally installed gems
```
